include/oneapi/dpl/pstl/iterator_impl.h

```diff
@@ -157,6 +190,22 @@ namespace oneapi
 {
 namespace dpl
 {
+
+template <typename T>
+struct is_indirectly_device_accessible_iterator
+    : decltype(oneapi::dpl::__internal::__is_onedpl_indirectly_device_accessible_iterator(std::declval<T>()))
+{
+    static_assert(
+        std::is_same_v<decltype(decltype(oneapi::dpl::__internal::__is_onedpl_indirectly_device_accessible_iterator(
+                           std::declval<T>()))::value),
+                       const bool>,
+        "Return type of is_onedpl_indirectly_device_accessible_iterator does not have the characteristics of a "
+        "bool_constant");
+};
+
+template <typename T>
+inline constexpr bool is_indirectly_device_accessible_iterator_v = is_indirectly_device_accessible_iterator<T>::value;
```

Comment:
Does it make sense to redefine it like
```suggestion
template <typename T>
using is_indirectly_device_accessible_iterator_t =
    std::integral_constant<bool, decltype(oneapi::dpl::__internal::__is_onedpl_indirectly_device_accessible_iterator(
                                     std::declval<T>()))::value>;
```